src/NUnitFramework/framework/Constraints/AttributeConstraint.cs

```diff
@@ -57,6 +58,10 @@ public AttributeConstraint(Type type, IConstraint baseConstraint)
         /// expected attribute and if its value matches the
         /// additional constraint specified.
         /// </summary>
+        /// <typeparam name="TActual"></typeparam>
+        /// <param name="actual">The value to be tested</param>
+        /// <exception cref="ArgumentException">Attribute was not found</exception>
```

Comment:
This is actually wrong. the argument is thrown if the provided Type is not an attribute.